### PR TITLE
Add unit tests for MavenResolver

### LINKED ISSUE #25075: Write unit tests for Maven Resolver
**Description:**
Add unit tests for MavenResolver using Mockito testing framework.

### PR DESCRIPTION
**Pull request by suganyasuven**
Fixes #25075

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/maven-resolver/src/test/java/org/ballerinalang/maven/MavenResolverUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+/**
+ * Unit Tests for {@link MavenResolver}.
+ *
+ *  @since 2.0.0
+ */
+public class MavenResolverUnitTest {
+    MavenResolver resolver = mock(MavenResolver.class);
+
+    @Test
```
Comment (pramodya1994) [inline misc/maven-resolver/src/test/java/org/ballerinalang/maven/MavenResolverUnitTest.java:40]: Shall we add descriptions for each test case?

File: misc/maven-resolver/src/test/java/org/ballerinalang/maven/MavenResolverUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit Tests for {@link MavenResolver}.
+ *
+ *  @since 2.0.0
+ */
+public class MavenResolverUnitTest {
+    MavenResolver resolver = mock(MavenResolver.class);
```
Comment (pramodya1994) [inline misc/maven-resolver/src/test/java/org/ballerinalang/maven/MavenResolverUnitTest.java:38]:
```suggestion
    private MavenResolver resolver = mock(MavenResolver.class);
```

File: misc/maven-resolver/src/test/java/org/ballerinalang/maven/MavenResolverUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit Tests for {@link MavenResolver}.
+ *
+ *  @since 2.0.0
+ */
+public class MavenResolverUnitTest {
+    private MavenResolver resolver = mock(MavenResolver.class);
```
Comment (azinneera) [inline misc/maven-resolver/src/test/java/org/ballerinalang/maven/MavenResolverUnitTest.java:38]: What is the class we are trying to test here? Isn't it the `MavenResolver`?

Comment (suganyasuven) [reply misc/maven-resolver/src/test/java/org/ballerinalang/maven/MavenResolverUnitTest.java:38]: Yes. This is to test the functions of MavenResolverClass.

Comment (azinneera) [reply misc/maven-resolver/src/test/java/org/ballerinalang/maven/MavenResolverUnitTest.java:38]: Then we shouldn't be mocking the same class. Mocking is used to mimic other classes, not the class under test.

**Pull request closed without merging**